tfhe/src/shortint/ciphertext/compressed_modulus_switched_ciphertext.rs

```diff
@@ -0,0 +1,184 @@
+use super::common::*;
+use super::Ciphertext;
+use crate::core_crypto::prelude::compressed_modulus_switched_lwe_ciphertext::CompressedModulusSwitchedLweCiphertext;
+use crate::core_crypto::prelude::{keyswitch_lwe_ciphertext, CiphertextModulusLog, LweCiphertext};
+use crate::shortint::engine::ShortintEngine;
+use crate::shortint::server_key::{apply_programmable_bootstrap, LookupTableOwned};
+use crate::shortint::{CarryModulus, MessageModulus, ServerKey};
+
+/// An object to store a ciphertext in little memory.
+/// Decompressing it requires a PBS
+///
+/// # Example
+///
+/// ```rust
+/// use tfhe::shortint::gen_keys;
+/// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+///
+/// // Generate the client key and the server key:
+/// let (cks, sks) = gen_keys(PARAM_MESSAGE_2_CARRY_2_KS_PBS);
+///
+/// let clear = 3;
+///
+/// let ctxt = cks.unchecked_encrypt(clear);
+///
+/// // Can be serialized in a smaller buffer
+/// let compressed_ct = sks.switch_modulus_and_compress(&ctxt);
+///
+/// let decompressed_ct = sks.decompress(&compressed_ct);
+///
+/// let dec = cks.decrypt(&decompressed_ct);
+///
+/// assert_eq!(clear, dec);
+/// ```
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct CompressedModulusSwitchedCiphertext {
+    compressed_modulus_switched_lwe_ciphertext: CompressedModulusSwitchedLweCiphertext<u64>,
+    degree: Degree,
+    message_modulus: MessageModulus,
+    carry_modulus: CarryModulus,
+    pbs_order: PBSOrder,
+}
+
+impl ServerKey {
+    /// Packs a ciphertext to have a smaller serialization size
```

Comment:
missed as well the first time around, let's keep impl blocks in server_key/ add a module if needed